### PR TITLE
Save job to mondodb on submit

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: cdmtaskservice/mongo.py [hunk 1]
```diff
@@ -105,6 +110,17 @@ async def get_image(self, name: str, digest: str | None = None, tag: str | None
             raise NoSuchImageError(f"No image {name} {err} exists in the system.")
         return models.Image.model_construct(**doc)
 
+    async def save_job(self, job: models.Job):
+        """ Save a job. Job IDs are expected to be unique."""
+        _not_falsy(job, "job")
+        jobd = job.model_dump()
+        jobi = jobd[models.FLD_JOB_JOB_INPUT]
```
Comment (Tianhao-Gu) [inline cdmtaskservice/mongo.py:117]:
```suggestion
        jobi = jobd.get(models.FLD_JOB_JOB_INPUT, {})
```

Can we assume the job input always exists at this stage?

File: cdmtaskservice/mongo.py [hunk 2]
```diff
@@ -105,6 +110,17 @@ async def get_image(self, name: str, digest: str | None = None, tag: str | None
             raise NoSuchImageError(f"No image {name} {err} exists in the system.")
         return models.Image.model_construct(**doc)
 
+    async def save_job(self, job: models.Job):
+        """ Save a job. Job IDs are expected to be unique."""
+        _not_falsy(job, "job")
```
Comment (Tianhao-Gu) [inline cdmtaskservice/mongo.py:115]: Do we need this check? I cannot think of a case when `not job` is `true`.

File: cdmtaskservice/mongo.py [hunk 3]
```diff
@@ -105,6 +110,17 @@ async def get_image(self, name: str, digest: str | None = None, tag: str | None
...
+        jobd = job.model_dump()
+        jobi = jobd[models.FLD_JOB_JOB_INPUT]
+        jobi[models.FLD_JOB_INPUT_RUNTIME] = jobi[models.FLD_JOB_INPUT_RUNTIME].total_seconds()
+        # don't bother checking for duplicate key exceptions since the service is supposed
+        # to ensure unique IDs
+        # TODO test that the timedelta is correctly reinstantiated in a get_job method
+        await self._col_jobs.insert_one(jobd)
```
Comment (Tianhao-Gu) [inline cdmtaskservice/mongo.py:122]: Not sure if it's worth to add a `try-catch` block here. DB client sometimes can generate error logs that are difficult to interpret.

**Review by Tianhao-Gu — APPROVED** [review]

File: cdmtaskservice/mongo.py (hunk 1 above)
Comment (MrCreosote) [reply cdmtaskservice/mongo.py:117]: Yes, it's a required part of a Job

File: cdmtaskservice/mongo.py (hunk 2 above)
Comment (MrCreosote) [reply cdmtaskservice/mongo.py:115]: If someone does `save_job(None)` or the equivalent. It shouldn't happen, but it ensures the inputs are valid

File: cdmtaskservice/mongo.py (hunk 3 above)
Comment (MrCreosote) [reply cdmtaskservice/mongo.py:122]: Do you mean exceptions or logs here? try/catch won't affect logs obviously.

A try / catch would just presumably wrap the mongo exception inside a different exception with a message like "Failed to connect to db" which is why I didn't add it - it doesn't seem very useful and winds up as a 500 at the API level regardless

If we find that it helps to wrap the exception in the future I'm fine with adding it but right now I'm not seeing much of a benefit

**Pull request merged**